tracks/ruby/exercises/accumulate/mentoring.md

```diff
@@ -1,32 +1,35 @@
 # Mentoring
 
-This exercise is asking you to re-implement `Array#map`. The goal is to get a
-feel for what's happening under the hood.
+This exercise is asking you to re-implement `Array#map`.
+The goal is to get a feel for what's happening under the hood.
 
 ## Reasonable Solutions
 
-There are 4 general classes of solutions that fall in a matrix
-`(functional | imperative) X ( high-level | low-level)`.
+There are 4 general classes of solutions that fall in a matrix `(functional | imperative) X ( high-level | low-level)`.
 
 ```ruby
 # High-level functional
 
 class Array
+
   def accumulate
     reduce([]) { |arr, item| arr.append(yield item) }
   end
+
 end
```

Comment:
Could we maybe revert the adding whitespace lines? I don't think it adds anything for these simple examples, but means you can fit less info on the screen when reading the notes.

Reply:
Except that the last method is not tightly coupled to the class definition nor the end of the class definition?  So "white space as meaning"?

I could convert all methods that are single line methods to:

```ruby
def accumulate= reduce([]) { |memo, item| meno.append(yield item) }
```

… if we are worried about viewing real estate rather than "space around the things".

But yeah, up to you.  I think it communicates the right thing.

Reply:
> Except that the last method is not tightly coupled to the class definition nor the end of the class definition

I mean, it's tightly coupled in the sense it's part of it. And surely by the same logic the code inside the method isn't tightly coupled either, so there should be whitespace added within the method.

I don't see how adding the whitespace makes things more readable for a mentor. The counter-example you give clearly reduces the ability to see what's going on, so that makes it less readable. But others may have different perspectives. Let's see if anyone else weighs in.

Reply:
Right.  Yeah, the white space in the methods was considered, but I went for the larger scope of that, treating the methods as units.  I would have personally added some white space in the method (and rearranged for organization as well) but it was too much.  So I left it to the smallest change (I considered) reasonable.